src/rpc/core_rpc_server.cpp

```diff
@@ -616,6 +616,23 @@ namespace cryptonote
       return true;
     }
 
+    unsigned int concurrency_count = std::thread::hardware_concurrency() * 4;
```

Comment:
I requested the scaling by 4, so that we only reject "obviously silly" arguments, since the number of threads is specified by the user, so I feel it makes more sense to trust it more.

--max-concurrency is a bodge I added to ward off a crash on Mac, which may be either a bug in Max kernel/libc, or monero, and I couldn't work it out. If you feel like double checking the usage of threads in the wallet2 refresh code, maybe you can find what's wrong :)